rust/agama-lib/src/store.rs

```diff
@@ -246,6 +266,20 @@ impl Store {
         Ok(())
     }
 
+    async fn detect_selected_product(&self) -> Result<bool, ProductStoreError> {
+        let product_client = ProductHTTPClient::new(self.http_client.clone());
+        let product = product_client.product().await?;
+        Ok(!product.is_empty())
+    }
+
+    fn check_selected_product(selected: bool) -> Result<(), StoreError> {
```

Comment:
I would prefer `ensure_selected_product` because it communicates better the intention.